### ISSUE TITLE
running dump() fails when decode_responses=True in a pipeline

### ISSUE THREAD

**Issue by brianmaissy** [redis/redis-py#1884]
This is an extension of issue #1254, which was partially (but I belive not fully) resolved in #1731.

I think that when non-decodable commands are called from within a pipeline, they are decoded anyway.

See my comment on the original issue here: https://github.com/redis/redis-py/issues/1254#issuecomment-1003683647

**Comment by dvora-h** [comment]
HI @brianmaissy 
This is a real bug, but however it doesn't make sense to use one of these commands in a pipeline, so for now we will leave it as an open bug.

**Comment by brianmaissy** [comment]
Why doesn't it make sense to use `dump()` in a pipeline? I do so in order to dump multiple keys.

**Comment by chayim** [comment]
@brianmaissy We should support this (and don't today), the best answer is to use a connection where deconde_responses is False.  While I realize this isn't ideal, this is the best current answer. 

 tl;dr yes a bug we should fix in the future, and labelled accordingly.

**Comment by brianmaissy** [comment]
Yeah that's the workaround I've been using. Thank you.

**Comment by chayim** [comment]
Since we added the NEVER_DECODE target as part of another bug - closing this off. It works now.

**Comment by brianmedigate** [comment]
in what version is it fixed?

**Comment by brianmedigate** [comment]
I'm using 4.3.4, but it still doesn't work when the non-decodable command is used in a pipeline

**Comment by brianmedigate** [comment]
@chayim @dvora-h can you please confirm that this is still open?

**Comment by dvora-h** [comment]
@brianmaissy Yes

**Comment by brianmedigate** [comment]
AFAIC this is still an issue

**Comment by petyaslavova** [comment]
Hi @brianmedigate the dump command works without issues if you don't run your pipeline in transaction mode (client.pipeline(transaction=False)). Would that work for your use case?

**Comment by brianmedigate** [comment]
Maybe. But shouldn't this be fixable regardless?
